test/suite/test_config12.py

```
@@ -0,0 +1,67 @@
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_config12.py
+#   Test the prealloc_init_count configuration.
+#
+
+import wttest
+from wiredtiger import stat
+from wtscenario import make_scenarios
+from wtdataset import SimpleDataSet
+import os, time
+
+class test_config12(wttest.WiredTigerTestCase):
+    prealloc_init_count = 3
+    conn_config = 'create,log=(enabled,prealloc=true,log_prealloc_init_count=' + str(prealloc_init_count) + ')'
+
+    # Gets the number of files under the directory that contain the specified prefix name.
+    def count_files_with_prefix(self, directory, prefix):
+        count = 0
+        for filename in os.listdir(directory):
+            if filename.startswith(prefix):
+                count += 1
+        return count
+
+    def test_config12(self):
+        uri = "table:config12"
+        # Create a very basic table.
+        ds = SimpleDataSet(self, uri, 10, key_format='S', value_format='S')
+        ds.populate()
+
+        retry_times = 0
+        # Wait for the prealloc file creation to finish, Wait a maximum of 60 seconds.
+        while (self.count_files_with_prefix(os.getcwd(), "WiredTigerPreplog.") < self.prealloc_init_count):
+            retry_times += 1
+            time.sleep(1)
+            if retry_times > 60:
+                raise Exception("Timed out waiting for creating prealloc file")
+
+        # Wait for 5 seconds and continue the check to prevent the number of WiredTigerPreplog being higher than prealloc_init_count.
```

What is the purpose of this part of the test? It is just a 5 second version of the loop and doesn't seem to add anything. If anything, the test really doesn't need the table creation and populate, does it? This change is for the initial pre-allocation. By not creating a table and data, then you can see the pre-allocation without adding a lot of records to the log files and consuming them.

---

I added a delay to ensure that the number of files created is 3 rather than 4 or more.

If prealloc_init_count is set to 3, the number of WiredTigerPreplog files in prealloc can only be 3 and cannot exceed 3.

Assuming we created 4 WiredTigerPreplog.xxxx files, we may ignored the exception case if we hadn't waited here and double-checked.